lib/vmpooler/pool_manager.rb

```diff
@@ -555,6 +555,75 @@ def check_pool(pool,
       end
     end
 
+    def prepare_template(pool, provider)
+      if $config[:config]['create_template_delta_disks']
+        # Ensure templates are evaluated for delta disk creation on startup
+        return if $redis.hget('vmpooler__config__updating', pool['name'])
+        unless $redis.hget('vmpooler__template__prepared', pool['name'])
```

Comment:
What hapens when I update the template say from -0.0.1 to -0.0.2? My understanding is this `unless` will skip because it's already been done for 0.0.1, which is not what we want

Reply:
In this case `update_pool_template` will handle the pool update, where the pool will be flushed, it will once again evaluate the template for delta prepare disk creation, and then allow pool repopulation with the new template. It's true that `prepare_template` will not handle this case. It's perhaps a little confusing, but it is by design so they can handle their processes separately.